### PR TITLE
Modify AccountTest to use bip39Entropy values

### PR DESCRIPTION
**Pull request by samdealy**
### Motivation
We added support for bip 39 entropy values, so we need to test that our SDK code is handling these values correctly.

In this PR
I copied the test data from the mobilecon repo into our androidTest resource directory (note that we can't use the "res" directory because the test runner can't find resources there). I then modified AccountTest to ensure that the AccountKey.fromMnemonicPhrase and AccountKey.fromBip39Entropy create the correct AccountKeys (i.e. the the test data values).

Ticket: [ANDROID-67 - Create new unit tests and improve existing unit tests](https://mobilecoin.atlassian.net/browse/ANDROID-67)

### Future Work
Continue adding unit tests.

### CODE REVIEW

File: android-sdk/src/androidTest/java/com/mobilecoin/lib/AccountTest.java [hunk 1]
```diff
@@ -104,46 +142,100 @@ public void test_deserialize() throws SerializationException, InvalidUriExceptio
...
-                fogConfig.getFogUri(),
-                fogConfig.getFogReportId(),
-                fogConfig.getFogAuthoritySpki()
-        );
+    public void fromMnemonicPhrase_createsCorrectPrivateKeys() throws Exception {
+        Assert.assertFalse(accountTestData.isEmpty());
+        for (AccountTestData accountTestData : accountTestData) {
+            String calculatedMnemonicPhrase =
+                    Mnemonics.bip39EntropyToMnemonic(accountTestData.bip39Entropy);
+            Assert.assertEquals(calculatedMnemonicPhrase, accountTestData.mnemonic);
```
Comment (voloshyn) [inline android-sdk/src/androidTest/java/com/mobilecoin/lib/AccountTest.java:152]: `assertEquals` treats the first argument as an "expected result", so if this test fails the tester will see an incorrect message, something like "expected value `XYZ`, but got `ABC`" while actually `ABC` is the correct value

File: android-sdk/src/androidTest/java/com/mobilecoin/lib/AccountTest.java [hunk 2]
```diff
@@ -104,46 +142,100 @@ public void test_deserialize() throws SerializationException, InvalidUriExceptio
...
+            this.bip39Entropy = bip39Entropy;
+            this.mnemonic = mnemonic;
+            this.accountIndex = accountIndex;
+            this.viewPrivateKey = viewPrivateKey;
+            this.spendPrivateKey = spendPrivateKey;
+        }
     }
- */
-}
+}
```
Comment (voloshyn) [inline android-sdk/src/androidTest/java/com/mobilecoin/lib/AccountTest.java:241]: missing the new line

**Review by voloshyn — APPROVED** [review]
I left a few comments to improve readability, other than LGTM

File: android-sdk/src/androidTest/java/com/mobilecoin/lib/AccountTest.java [hunk 3]
```diff
@@ -61,6 +75,30 @@
...
+        String line;
+        List<AccountTestData> accountTestData = new ArrayList<>();
+        try (BufferedReader reader =
+                     new BufferedReader(new InputStreamReader(accountTestDataStream))) {
+            while ((line = reader.readLine()) != null) {
+                JSONObject mainObject = new JSONObject(line);
+                accountTestData.add(AccountTestData.fromJsonObject(mainObject));
+            }
+        } catch (IOException | JSONException e) {
+            Log.e("ACCOUNT_TEST", e.getMessage());
```
[the remark this answers is not in the document]
Comment (samdealy) [reply android-sdk/src/androidTest/java/com/mobilecoin/lib/AccountTest.java:96]: Done.

File: android-sdk/src/androidTest/java/com/mobilecoin/lib/AccountTest.java [hunk 4]
```diff
@@ -104,46 +142,100 @@ public void test_deserialize() throws SerializationException, InvalidUriExceptio
...
-        AccountKey accountKey = AccountKey.fromMnemonicPhrase(
-                mnemonic,
-                0,
-                fogConfig.getFogUri(),
-                fogConfig.getFogReportId(),
-                fogConfig.getFogAuthoritySpki()
-        );
+    public void fromMnemonicPhrase_createsCorrectPrivateKeys() throws Exception {
+        Assert.assertFalse(accountTestData.isEmpty());
+        for (AccountTestData accountTestData : accountTestData) {
```
[the remark this answers is not in the document]
Comment (samdealy) [reply android-sdk/src/androidTest/java/com/mobilecoin/lib/AccountTest.java:149]: Done.

File: android-sdk/src/androidTest/java/com/mobilecoin/lib/AccountTest.java (hunk 1 above)
Comment (samdealy) [reply android-sdk/src/androidTest/java/com/mobilecoin/lib/AccountTest.java:152]: Done.

**Pull request merged**